### PR TITLE
tests: Add detailed meta-test for CastFullyTypedFormat.

### PR DESCRIPTION
**Pull request by HansKristian-Work**
Signed-off-by: Hans-Kristian Arntzen <post@arntzen-software.no>

### CODE REVIEW

File: tests/d3d12_descriptors.c [hunk 1]
```diff
@@ -4469,3 +4469,396 @@ void test_view_min_lod(void)
...
+     * Should also serve as more clear documentation on actual behavior. */
+    D3D12_FEATURE_DATA_D3D12_OPTIONS3 options3;
+    D3D12_UNORDERED_ACCESS_VIEW_DESC uav;
+    D3D12_SHADER_RESOURCE_VIEW_DESC srv;
+    struct test_context context;
+    ID3D12Resource *uav_texture;
+    ID3D12DescriptorHeap *heap;
+    ID3D12Resource *texture;
+    unsigned int i;
+    HRESULT hr;
```
Comment (doitsujin) [inline tests/d3d12_descriptors.c:4486]: unused

**Review by doitsujin — COMMENTED** [review]
some warning, looks good otherwise

File: tests/d3d12_descriptors.c [hunk 2]
```diff
@@ -4707,3 +4707,158 @@ void test_typed_srv_uav_cast(void)
...
+    D3D12_CLEAR_VALUE clear_value;
+    struct test_context context;
+    unsigned int test_iteration;
+    ID3D12DescriptorHeap *heap;
+    D3D12_RESOURCE_DESC desc;
+    ID3D12Resource *texture;
+    unsigned int i, j;
+    FLOAT colors[4];
+    bool fast_clear;
+    HRESULT hr;
```
Comment (doitsujin) [inline tests/d3d12_descriptors.c:4726]: unused again

File: tests/d3d12_descriptors.c (hunk 1 above)
Comment (HansKristian-Work) [reply tests/d3d12_descriptors.c]: Fixed.

File: tests/d3d12_descriptors.c (hunk 2 above)
Comment (HansKristian-Work) [reply tests/d3d12_descriptors.c]: Fixed.

**Pull request merged**